moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -993,6 +993,10 @@ bool TimeOptimalTrajectoryGeneration::computeTimeStamps(robot_trajectory::RobotT
 
   // Compute sample count
   size_t sample_count = std::ceil(parameterized.getDuration() / resample_dt_);
+  if (sample_count < num_points)
```

Comment:
By increasing `sample_count` to `num_points` the trajectory is filled with duplicate trajectory points with the same time stamps at the end which breaks CI (ros2_control isn't happy about that). The reason is that the computed timestamps in the loop below exceed the trajectory duration (see https://github.com/ros-planning/moveit2/pull/468/files#diff-227cef8a99e7eaf48f74355541c0f673279e65839065e5f29646319063377cdeR1008). Is it necessary for you to have at least `num_points` waypoints? If so, you would have to recompute `resample_dt_` as `parameterized.getDuration() / num_points` to match the trajectory duration. But I think the better option would be to just specify a resampling step that works in general and remove this change.

Reply:
> But I think the better option would be to just specify a resampling step that works in general and remove this change.

Do you mean an entirely separate PlanningRequestAdapter that does the resampling? Because I agree, but don't know how to move forward with that in a way that doesn't break everyone's existing configurations.

Reply:
No, I meant setting the resampling_dt parameter small enough so that you don't have to worry about trajectory waypoint count at all.